core/domain/rights_manager.py

```diff
@@ -873,7 +902,15 @@ def _deassign_role(committer, removed_user_id, activity_id, activity_type):
     committer_id = committer.user_id
     activity_rights = _get_activity_rights(activity_type, activity_id)
 
-    if not check_can_modify_activity_roles(committer, activity_rights):
+    user_can_modify_activity_roles = (
+        check_can_modify_activity_roles(committer, activity_rights) or (
+            check_can_manage_voice_artist_in_activity(
+                committer, activity_rights) and (
+                    activity_rights.is_voice_artist(removed_user_id)
```

Comment:
I am a bit confused by the lack of parity here. How come, for voice artist, we need to check that the removed_user_id is a voice artist, but we don't have to do that same check for other roles?